packages/drivers/routerlicious-driver/src/wholeSummaryDocumentStorageService.ts

```diff
@@ -275,33 +316,35 @@ export class WholeSummaryDocumentStorageService implements IDocumentStorageServi
 			},
 		);
 
-		assert(
-			normalizedWholeSummary.snapshotTree.id !== undefined,
-			0x275 /* "Root tree should contain the id" */,
-		);
-		const wholeFlatSummaryId: string = normalizedWholeSummary.snapshotTree.id;
-		const snapshotTreeVersion = {
-			id: wholeFlatSummaryId,
-			snapshotTree: normalizedWholeSummary.snapshotTree,
-		};
+		return normalizedWholeSummary;
+	}
 
+	private async initializeFromSnapshot(
+		normalizedWholeSummary: INormalizedWholeSummary,
+		versionId: string | null,
+	): Promise<string> {
+		const wholeFlatSummaryId = normalizedWholeSummary.snapshotTree.id;
+		assert(wholeFlatSummaryId !== undefined, 0x275 /* "Root tree should contain the id" */);
 		const cachePs: Promise<any>[] = [
-			this.snapshotTreeCache.put(this.getCacheKey(wholeFlatSummaryId), snapshotTreeVersion),
```

Comment:
It feels like we will update cache even if we loaded from cache. Is that the case? If so, feels like writing to cache should be inside of getVersions()

Reply:
It is inside getVersions only. So, once we return the versions after fetching the latest snapshot in getVersions we supply back the versionIds which is used by loader to actually make the snapshot call with the particular version where it is will used. This is same as odsp where we have inmemory cache where we cache trees which will be used later.

Reply:
Also the cache could have docid/latest but not docid/versionid, so that is why we are caching both. However before always doing put, I can do get and then if no result, then do put.